### PR TITLE
Use storage.k8s.io/v1 in tests instead of v1beta1

### PR DESCRIPTION
**Pull request by jsafrane**
This is trimmed version of #42477 and contains only tests of the new storage API. Together with #43285 it passes all dynamic provisioning tests on my GCE.

I did not change vsphere_utils.go and vsphere_volume_diskformat.go as @divyenpatel runs master vsphere tests with Kubernetes 1.5 - @divyenpatel, did I get it right?

@kubernetes/sig-storage-pr-reviews, @msau42, @ethernetdan
```release-note
NONE
```

### CODE REVIEW

**Comment by divyenpatel** [comment]
Thanks @jsafrane. Looks good.
We are in process of back porting tests to 1.5 branch.

File: test/e2e/volume_provisioning.go [hunk 1]
```diff
@@ -104,46 +105,44 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
...
-					testDynamicProvisioning(c, claim, requestedSize)
-				} else {
-					// Expected size of the volume is 2GiB, because the other three supported cloud
-					// providers allocate volumes in 1GiB chunks.
-					testDynamicProvisioning(c, claim, "2Gi")
-				}
-			})
+	framework.KubeDescribe("DynamicProvisioner", func() {
+		It("should create and delete persistent volumes [Slow] [Volume]", func() {
+			framework.SkipUnlessProviderIs("openstack", "gce", "aws", "gke")
```
Comment (msau42) [inline test/e2e/volume_provisioning.go]: @divyenpatel You should add vsphere back to all these tests once your CI environment is updated.

Comment (divyenpatel) [inline test/e2e/volume_provisioning.go]: @msau42 Yes sure.

cc: @BaluDontu @tusharnt

**Pull request merged**